src/selectors.ts

```diff
@@ -237,179 +248,91 @@ const generateSiteIntervalPreset = (soilData: SoilData) => {
   }
 };
 
-export const selectProjectSettings = createSelector(
-  [
-    (state: SharedState, projectId: string) =>
-      state.soilId.projectSettings[projectId],
-  ],
-  (projectSettings: ProjectSoilSettings) => {
-    if (!projectSettings) {
-      return undefined;
-    }
-    const depthIntervals = generateProjectIntervals(projectSettings) || [];
-    return { ...projectSettings, depthIntervals };
-  },
-);
+const useProjectSoilSettingsBase = <ID extends string | undefined>(
+  projectId: ID,
+): ID extends undefined ? undefined : ProjectSoilSettings => {
+  const projectSettings = useSelector(state =>
+    projectId === undefined
+      ? undefined
+      : state.soilId.projectSettings[projectId] ?? DEFAULT_PROJECT_SETTINGS,
```

Comment:
suggestion: this is a bit confused. please add parentheses to indicate precedence more explicitly.

Reply:
huh the formatter actually refuses to allow parentheses, i'll rewrite as an if statement because i agree. i like expression-based over statement-based syntax but i admit i shoehorn it into languages that don't support it well a bit too much sometimes...